pkg/controllers/clusteroperator_controller.go

```diff
@@ -97,6 +98,19 @@ func (r *CloudOperatorReconciler) Reconcile(ctx context.Context, _ ctrl.Request)
 		return ctrl.Result{}, err
 	}
 
+	clusterProxy := &configv1.Proxy{}
+	if err := r.Get(ctx, client.ObjectKey{Name: proxyResourceName}, clusterProxy); errors.IsNotFound(err) {
+		clusterProxy = &configv1.Proxy{}
```

Comment:
Why is this initialisation required? I think line 101 should be sufficient to ensure it's not a nil later, and `Get` will only modify the object if get succeeds, so I don't think this is actually required is it?